darwin/dataset/remote_dataset.py

```diff
@@ -100,6 +100,9 @@ def push(
             )
             if not files_to_upload:
                 raise NotFound("No files to upload, check your path and exclusion filters")
+        elif extensions_to_exclude is not None:
```

Comment:
I think `extensions_to_exclude` is an optional argument for `find_files()`, so perhaps we can avoid this `elif` branch and simply do
```python
files_to_upload = find_files(files_list=files_to_upload, exclude=extensions_to_exclude)
```